backend/fetcher.py

```diff
@@ -160,193 +159,156 @@ def fetch_spec_file(self) -> Optional[dict[str, str]]:
 
 class KojiProvider(RPMProvider):
     koji_url = "https://koji.fedoraproject.org"
-    logs_to_look_for = ["build.log", "root.log", "mock_output.log"]
+    # checkout.log - for dist-git repo cloning problems
+    logs_to_look_for = ["build.log", "root.log", "mock_output.log",
+                        "checkout.log"]
     koji_pkgs_url = "https://kojipkgs.fedoraproject.org/work"
 
     def __init__(self, build_or_task_id: int, arch: str) -> None:
-        self.build_or_task_id = build_or_task_id
-        self.arch = arch
         api_url = "{}/kojihub".format(self.koji_url)
         self.client = koji.ClientSession(api_url)
 
-    @cached_property
-    def _is_build_id(self) -> bool:
+        self.arch = arch
+        self.task_id = None
+        self.build_id = None
+        # this block detects what we got: is it build or task?
+        # failed builds are useless sadly, we will only work with tasks
         try:
-            return self.client.getBuild(self.build_or_task_id) is not None
+            self.build = self.client.getBuild(build_or_task_id)
         except koji.GenericError:
-            return False
-
-    def _fetch_build_logs_from_build_id(self) -> list[dict[str, str]]:
-        """
-        Obtain build logs from a failed build by traversing
-        through the task hierarchy to the buildArch task's logs
-
-        Koji's method getBuildLogs sadly only works with successful builds.
-        """
-        koji_build = self.client.getBuild(self.build_or_task_id)
-        root_task_id = koji_build['task_id']
-        # the response of getTaskDescendents:
-        #   {'112162296': [{'arch': 'noarch', 'awaited': False...
-        task_descendants = self.client.getTaskDescendents(root_task_id)[str(root_task_id)]
-        for task_info in task_descendants:
-            if task_info['arch'] == self.arch and task_info['method'] == 'buildArch' \
-                    and task_info['state'] == 5:
-                # this is the one and only ring!
-                self.build_or_task_id = task_info['id']
-                return self._fetch_task_logs_from_task_id()
-        return []
+            # great, no need to take care of builds
+            self.build = None
+
+        if self.build:
+            self.build_id = build_or_task_id
+            # it's a build, we need to find the right task now
+            root_task_id = self.build['task_id']
+            # the response of getTaskDescendents:
+            #   {'112162296': [{'arch': 'noarch', 'awaited': False...
+            task_descendants = self.client.getTaskDescendents(root_task_id)[str(root_task_id)]
+            for task_info in task_descendants:
+                if task_info['arch'] == arch \
+                        and task_info['method'] in ("buildArch", "buildSRPMFromSCM") \
+                        and task_info['state'] == 5:
+                    # this is the one and only ring!
+                    self.task_id = task_info['id']
+                    break
```

Comment:
Nice! This is a bit OT but I have a question here - I recently noticed that fetching from packit (koji case) doesn't work because we need to look for task descendants. My first idea for the solution was the same as yours here - go for descendants and use the first task passing the if statement. But what if there were more failed tasks? What to do then? if this does not matter we could reuse this code in else statement if the task id has descendants to fix packit-koji issue.

Reply:
that is a good point

I started typing a response and realized...

if more tasks failed, it means they will have a different arch, and we already ask users that - the arch that we should select; the SRPM build task has a different method name than buildArch

I think this should always pick only one task though I'm no koji expert